File: Data Structures & Algorithms/max-water-container/submission-3.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {

        int right = height.size()-1;

        int left = 0;

        int maxArea = 0;

        int area=0;


 

        while (left != right){

            area = (right - left) * min(height[left], height[right]);

            maxArea = area>maxArea ? area:maxArea;

 

            if( height[right]> height[left])

            {

                left++;

            }

            else{right--;}

        }

 

        return maxArea;

    }
};

File: Data Structures & Algorithms/is-palindrome/submission-3.cs
public class Solution {
    public bool IsPalindrome(string s) {
        int n = s.Length;
        int left = 0;
        int right = n-1;

        while (left<right){
            while(left<right && !char.IsLetterOrDigit(s[left])) left++;
            while(left<right && !char.IsLetterOrDigit(s[right])) right--;

            if(char.ToLower(s[left]) == char.ToLower(s[right])){
                left++;
                right--;
            }
            else return false;
        }

        return true;

        
    }
}
